models/business_models/product/newtab/newtab_visits_v1.sql

```diff
@@ -0,0 +1,336 @@
+{{ 
+   config(
+        materialized='incremental',
+        partition_by = { 'field': 'submission_date', 'data_type': 'date' },
+        incremental_strategy = 'insert_overwrite',
+        require_partition_filter = true,
+        cluster_by = ["channel", "country_code"]
+    )
+}}
+
+
+
+WITH events_unnested AS (
+  SELECT
+    DATE(submission_timestamp) AS submission_date,
+    category AS event_category,
+    name AS event_name,
+    timestamp AS event_timestamp,
+    client_info,
+    metadata,
+    normalized_os,
+    normalized_os_version,
+    normalized_country_code,
+    normalized_channel,
+    ping_info,
+    extra AS event_details,
+    metrics
+  FROM
+    -- https://dictionary.telemetry.mozilla.org/apps/firefox_desktop/pings/newtab
+    `moz-fx-data-shared-prod.firefox_desktop_stable.newtab_v1`,
+    UNNEST(events)
+  WHERE
+    category IN ('newtab', 'topsites', 'newtab.search', 'newtab.search.ad', 'pocket')
+    AND name IN (
+      'closed',
+      'opened',
+      'impression',
+      'issued',
+      'click',
+      'save',
+      'topic_click',
+      'dismiss'
+    )
+    AND DATE(submission_timestamp) > "2024-03-30" -- limit backfill
+    {% if is_incremental() %}
+    AND DATE(submission_timestamp) > (SELECT max(submission_date) FROM {{ this }} WHERE submission_date > '2020-01-01')
```

Comment:
Out of curiosity, why the manual filter on `submission_date > '2020-01-01'`?

Reply:
The table has a required partition filter on `submission_date`, so in order for the query to run `submission_date` needs to be filtered on something